src/App.tsx

```diff
@@ -1,61 +1,141 @@
+import { ChangeEvent, FormEvent, useState } from 'react';
+
+import { FormFields } from './types/Form';
+import { User } from './types/User';
+
+import { validateForm } from './utils/validateForm';
+import { getUserById } from './utils/getUserById';
+
+import {
+  initialFormErrors,
+  initialFormValues,
+} from './constants/initialFormState';
+
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const initialTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId, usersFromServer),
+}));
 
 export const App = () => {
+  const [count, setCount] = useState(0);
+  const [todos, setTodos] = useState(initialTodos);
+  const [formData, setFormData] = useState(initialFormValues);
+  const [formErrors, setFormErrors] = useState(initialFormErrors);
+  const [user, setUser] = useState<User>();
+
+  const addTodo = () => {
+    const newTodo = {
+      id: Math.max(...todos.map(todo => todo.id)) + 1,
+      completed: false,
+      user,
+      ...formData,
+    };
+
+    setTodos(prev => [...prev, newTodo]);
+  };
+
+  const reset = () => {
+    setFormData(initialFormValues);
+    setFormErrors(initialFormErrors);
+  };
+
+  const handleChange = (
+    event: ChangeEvent<HTMLInputElement | HTMLSelectElement>,
+  ) => {
+    const { name, value } = event.target;
+
+    setFormData(prev => ({ ...prev, [name]: value }));
+
+    if (name === FormFields.USER_ID) {
+      const selectedUser = getUserById(Number(value), usersFromServer);
+
+      setUser(selectedUser);
+    }
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    const errors = validateForm(formData);
+
+    setFormErrors(errors);
+    const isFromValid = Object.keys(errors).length === 0;
+
+    if (isFromValid) {
+      addTodo();
+      reset();
+      setCount(count + 1);
```

Comment:
`use functional setState`: Using functional setState helps make sure you're updating the state based on the latest value, avoiding problems that can happen if the state changes while ur code is running.
```suggestion
      setCount((prevCount) => prevCount + 1);
```

Reply:
Done)
Thx for the improvement comments🔥